packages/health-check/src/checks/critical.js

```diff
@@ -63,6 +80,25 @@ async function uploadCheck(done) {
   done({ name: "upload_file", time: calculateElapsedTime(time), ...data });
 }
 
+// skylinkHealthCheck checks if the skylink has reached full redundancy
+async function skylinkHealthCheck(skylink, numRetries, authCookie) {
+  try {
+    const response = await got(`https://${process.env.PORTAL_DOMAIN}/skynet/health/skylink/${skylink}`, {
+      headers: { cookie: authCookie },
+    });
+    const healthData = getResponseContent(response);
+    if (healthData.basesectorredundancy !== 10 && numRetries > 0) {
+      return skylinkHealthCheck(skylink, numRetries - 1);
+    }
+    if (healthData.basesectorredundancy !== 10 && numRetries === 0) {
+      throw "Skylink did not reach full redundancy";
+    }
+    return response;
+  } catch (error) {
+    throw error;
```

Comment:
if you are just rethrowing the error, you can skip the try-catch block altogether

Reply:
good to know 👍